tests/phpunit/elementor/schemas/test-usage.php

```diff
@@ -0,0 +1,108 @@
+<?php
+namespace Elementor\Tests\Phpunit\Schemas;
+
+use Elementor\Core\Utils\ImportExport\WP_Import;
+use Elementor\Testing\Elementor_Test_Base;
+use Elementor\Tracker;
+use JsonSchema\Exception\ValidationException;
+use JsonSchema\SchemaStorage;
+use JsonSchema\Validator;
+
+
+class Test_Usage extends Elementor_Test_Base {
+	const HTTP_USER_AGENT = 'test-agent';
+
+	public function setUp() {
+		parent::setUp();
+
+		// Required by `Tracker::get_tracking_data`.
+		$_SERVER['HTTP_USER_AGENT'] = self::HTTP_USER_AGENT;
+	}
+
+	protected function run_validation_against_schema( $data ) {
+		// Since the usage system represents objects as array instead of stdClass.
+		$data = json_decode( json_encode( $data ) );
+
+		// Validate
+		$validator = new Validator;
+		$validator->validate($data, (object) ['$ref' => 'file://' . ELEMENTOR_PATH . 'schemas/usage.json'] );
+
+		if ( ! $validator->isValid() ) {
+			$error_message = 'JSON does not validate. Violations:' . PHP_EOL;
+			foreach ( $validator->getErrors() as $error ) {
+				$error_message .= sprintf( '[%s] %s' . PHP_EOL, $error['property'], $error['message'] );
+			}
+
+			throw new ValidationException( $error_message );
+		}
+
+		return true;
+	}
+
+	protected function validation_current_tracking_data_against_schema() {
+		return $this->run_validation_against_schema( Tracker::get_tracking_data() );
+	}
+
+	public function test__ensure_clean_is_valid() {
+		// Act + Assert.
+		$this->assertTrue( $this->validation_current_tracking_data_against_schema() );
+	}
+
+	public function test__ensure_invalid_exception() {
+		// Arrange.
+		$this->expectException( ValidationException::class );
+
+		// Assert.
+		$this->run_validation_against_schema( [] );
+	}
+
+	public function test__ensure_all_objects_have_no_additional_properties() {
+		// Arrange.
+		$json_schema_object = json_decode( file_get_contents( ELEMENTOR_PATH . 'schemas/usage.json' ) );
+
+		$schema_storage = new SchemaStorage();
+		$schema_storage->addSchema('usage', $json_schema_object);
+
+		$properties_all_objects_recursive = function( $node, callable $callback ) use ( &$properties_all_objects_recursive ) {
+			if ( ! ( $node instanceof \stdClass ) ) {
+				return;
+			}
+
+			if ( isset( $node->properties ) || isset( $node->patternProperties ) ) {
+				$callback( $node );
+			}
+
+			foreach ( $node as $part ) {
+				$properties_all_objects_recursive( $part, $callback );
+			}
+		};
+
+		// Act.
+		$usage_schema = $schema_storage->getSchema( 'usage' );
+
+		// Assert.
+		$properties_all_objects_recursive( $usage_schema, function ( $node ) {
+			$id = $node->{'$id'};
+			$this->assertTrue( isset( $node->additionalProperties ),
+				"Ensure node: '$id' 'additionalProperties' exists" );
```

Comment:
Why? assertFalse is not enough?

Reply:
It is more for debugging propose when it fails without this line it will show something like this:
`Undefined property: stdClass::$additionalProperties`

but with the line it will show `Ensure node: '#/system/plugins/active_plugins/$plugin' 'additionalProperties' exists`